packages/react-html/README.md

```diff
@@ -12,112 +12,267 @@ $ yarn add @shopify/react-html
 
 ## Usage
 
-The `<HTML>` component serves as a top level wrapper for a react application, allowing you to avoid needing any kind of server side template, in favor of purely using `reactDom.renderToString`.
+This PR exposes two entrypoints:
 
-```javascript
-import * as React from 'react';
-import {renderToString} from 'react-dom/server';
+- `@shopify/react-html`: which contains code that can be used on server and client. Code in your `app` and `client` directories should import from this entrypoint.
+- `@shopify/react-html/server`: which contains all of `react-html`, but also includes some features that are not safe to import on the client. Code in your `server` directory should import from this entrypoint.
 
-import HTML, {DOCTYPE} from '@shopify/react-html';
-import MyApp from '../app';
+### App code
 
-export default (ctx, next) => {
-  // we have to prepend DOCTYPE to serve valid HTML
-  ctx.body = DOCTYPE + renderToString(
-    <HTML>
-      <MyApp />
-    </HTML>
-  );
+The main entry point of this library is primarily concerned with giving you the ability to [serialize and deserialize information](https://github.com/Shopify/web-foundation/blob/master/Proposals/02%20-%20Serialization%20in%20application%20code.md) from within your application code. There are three parts to this: a `Manager`, which stores and looks up serializations, a `Provider`, that distributes this manager throughout the application, and a function for generating type-safe serialization/ deserialization components.
```

Comment:
This sounds a bit too technical to me. Have you considered including a brief summary of the proposal in the readme here? Also, perhaps introduce some sort of terminology for the `function for generating type-safe serialization/ deserialization components` or its output?